File: Examples/test-suite/java/nested_class_runme.java

import nested_class.*;

public class nested_class_runme {

  static {
    try {
	System.loadLibrary("nested_class");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) {
    Outer outer = new Outer();
    SWIGTYPE_p_Outer__InnerStruct1 is1 = outer.makeInnerStruct1();
    SWIGTYPE_p_Outer__InnerClass1 ic1 = outer.makeInnerClass1();
    SWIGTYPE_p_Outer__InnerUnion1 iu1 = outer.makeInnerUnion1();

    SWIGTYPE_p_Outer__InnerStruct2 is2 = outer.makeInnerStruct2();
    SWIGTYPE_p_Outer__InnerClass2 ic2 = outer.makeInnerClass2();
    SWIGTYPE_p_Outer__InnerUnion2 iu2 = outer.makeInnerUnion2();

    SWIGTYPE_p_Outer__InnerClass4Typedef ic4 = outer.makeInnerClass4Typedef();
    SWIGTYPE_p_Outer__InnerStruct4Typedef is4 = outer.makeInnerStruct4Typedef();
    SWIGTYPE_p_Outer__InnerUnion4Typedef iu4 = outer.makeInnerUnion4Typedef();

    SWIGTYPE_p_Outer__InnerClass5 ic5 = outer.makeInnerClass5();
    SWIGTYPE_p_Outer__InnerStruct5 is5 = outer.makeInnerStruct5();
    SWIGTYPE_p_Outer__InnerUnion5 iu5 = outer.makeInnerUnion5();

    ic5 = outer.makeInnerClass5Typedef();
    is5 = outer.makeInnerStruct5Typedef();
    iu5 = outer.makeInnerUnion5Typedef();

    {
      SWIGTYPE_p_Outer__InnerMultiple im1 = outer.getMultipleInstance1();
      SWIGTYPE_p_Outer__InnerMultiple im2 = outer.getMultipleInstance2();
      SWIGTYPE_p_Outer__InnerMultiple im3 = outer.getMultipleInstance3();
      SWIGTYPE_p_Outer__InnerMultiple im4 = outer.getMultipleInstance4();
    }

    {
      SWIGTYPE_p_Outer__InnerMultipleDerived im1 = outer.getMultipleDerivedInstance1();
      SWIGTYPE_p_Outer__InnerMultipleDerived im2 = outer.getMultipleDerivedInstance2();
      SWIGTYPE_p_Outer__InnerMultipleDerived im3 = outer.getMultipleDerivedInstance3();
      SWIGTYPE_p_Outer__InnerMultipleDerived im4 = outer.getMultipleDerivedInstance4();
    }

    {
      SWIGTYPE_p_Outer__InnerMultipleDerived im1 = outer.getMultipleDerivedInstance1();
      SWIGTYPE_p_Outer__InnerMultipleDerived im2 = outer.getMultipleDerivedInstance2();
      SWIGTYPE_p_Outer__InnerMultipleDerived im3 = outer.getMultipleDerivedInstance3();
      SWIGTYPE_p_Outer__InnerMultipleDerived im4 = outer.getMultipleDerivedInstance4();
    }

    {
      SWIGTYPE_p_Outer__InnerMultipleAnonTypedef1 mat1 = outer.makeInnerMultipleAnonTypedef1();
      SWIGTYPE_p_Outer__InnerMultipleAnonTypedef2 mat2 = outer.makeInnerMultipleAnonTypedef2();
      SWIGTYPE_p_Outer__InnerMultipleAnonTypedef3 mat3 = outer.makeInnerMultipleAnonTypedef3();

      SWIGTYPE_p_Outer__InnerMultipleNamedTypedef mnt = outer.makeInnerMultipleNamedTypedef();
      SWIGTYPE_p_Outer__InnerMultipleNamedTypedef mnt1 = outer.makeInnerMultipleNamedTypedef1();
      SWIGTYPE_p_Outer__InnerMultipleNamedTypedef mnt2 = outer.makeInnerMultipleNamedTypedef2();
      SWIGTYPE_p_p_Outer__InnerMultipleNamedTypedef mnt3 = outer.makeInnerMultipleNamedTypedef3();
    }
    {
      SWIGTYPE_p_Outer__InnerSameName isn = outer.makeInnerSameName();
    }
  }
}

File: Examples/test-suite/kotlin/nested_class_runme.kt
@file:JvmName("nested_class_runme")

import nested_class.*

fun main() {
    try {
        System.loadLibrary("nested_class")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    val outer = Outer()
    @Suppress("UNUSED_VARIABLE")
    val is1: Outer.InnerStruct1? = outer.makeInnerStruct1()
    @Suppress("UNUSED_VARIABLE")
    val ic1: Outer.InnerClass1? = outer.makeInnerClass1()
    @Suppress("UNUSED_VARIABLE")
    val iu1: Outer.InnerUnion1? = outer.makeInnerUnion1()

    @Suppress("UNUSED_VARIABLE")
    val is2: Outer.InnerStruct2? = outer.makeInnerStruct2()
    @Suppress("UNUSED_VARIABLE")
    val ic2: Outer.InnerClass2? = outer.makeInnerClass2()
    @Suppress("UNUSED_VARIABLE")
    val iu2: Outer.InnerUnion2? = outer.makeInnerUnion2()

    @Suppress("UNUSED_VARIABLE")
    val ic4: Outer.InnerClass4Typedef? = outer.makeInnerClass4Typedef()
    @Suppress("UNUSED_VARIABLE")
    val is4: Outer.InnerStruct4Typedef? = outer.makeInnerStruct4Typedef()
    @Suppress("UNUSED_VARIABLE")
    val iu4: Outer.InnerUnion4Typedef? = outer.makeInnerUnion4Typedef()

    @Suppress("ASSIGNED_BUT_NEVER_ACCESSED_VARIABLE")
    var ic5: Outer.InnerClass5Typedef? = outer.makeInnerClass5()
    @Suppress("ASSIGNED_BUT_NEVER_ACCESSED_VARIABLE")
    var is5: Outer.InnerStruct5Typedef? = outer.makeInnerStruct5()
    @Suppress("ASSIGNED_BUT_NEVER_ACCESSED_VARIABLE")
    var iu5: Outer.InnerUnion5Typedef? = outer.makeInnerUnion5()

    @Suppress("UNUSED_VALUE")
    ic5 = outer.makeInnerClass5Typedef()
    @Suppress("UNUSED_VALUE")
    is5 = outer.makeInnerStruct5Typedef()
    @Suppress("UNUSED_VALUE")
    iu5 = outer.makeInnerUnion5Typedef()

    run {
        @Suppress("UNUSED_VARIABLE")
        val im1: Outer.InnerMultiple? = outer.MultipleInstance1
        @Suppress("UNUSED_VARIABLE")
        val im2: Outer.InnerMultiple? = outer.MultipleInstance2
        @Suppress("UNUSED_VARIABLE")
        val im3: Outer.InnerMultiple? = outer.MultipleInstance3
        @Suppress("UNUSED_VARIABLE")
        val im4: Outer.InnerMultiple? = outer.MultipleInstance4
    }

    run {
        @Suppress("UNUSED_VARIABLE")
        val im1: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance1
        @Suppress("UNUSED_VARIABLE")
        val im2: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance2
        @Suppress("UNUSED_VARIABLE")
        val im3: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance3
        @Suppress("UNUSED_VARIABLE")
        val im4: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance4
    }

    run {
        @Suppress("UNUSED_VARIABLE")
        val im1: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance1
        @Suppress("UNUSED_VARIABLE")
        val im2: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance2
        @Suppress("UNUSED_VARIABLE")
        val im3: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance3
        @Suppress("UNUSED_VARIABLE")
        val im4: Outer.InnerMultipleDerived? = outer.MultipleDerivedInstance4
    }

    run {
        @Suppress("UNUSED_VARIABLE")
        val mat1: Outer.InnerMultipleAnonTypedef1? = outer.makeInnerMultipleAnonTypedef1()
        @Suppress("UNUSED_VARIABLE")
        val mat2: Outer.InnerMultipleAnonTypedef1? = outer.makeInnerMultipleAnonTypedef2()
        @Suppress("UNUSED_VARIABLE")
        val mat3: SWIGTYPE_p_p_Outer__InnerMultipleAnonTypedef1? = outer.makeInnerMultipleAnonTypedef3()

        @Suppress("UNUSED_VARIABLE")
        val mnt: Outer.InnerMultipleNamedTypedef1? = outer.makeInnerMultipleNamedTypedef()
        @Suppress("UNUSED_VARIABLE")
        val mnt1: Outer.InnerMultipleNamedTypedef1? = outer.makeInnerMultipleNamedTypedef1()
        @Suppress("UNUSED_VARIABLE")
        val mnt2: Outer.InnerMultipleNamedTypedef1? = outer.makeInnerMultipleNamedTypedef2()
        @Suppress("UNUSED_VARIABLE")
        val mnt3: SWIGTYPE_p_p_Outer__InnerMultipleNamedTypedef? = outer.makeInnerMultipleNamedTypedef3()
    }
    run {
        @Suppress("UNUSED_VARIABLE")
        val isn: Outer.InnerSameName? = outer.makeInnerSameName()
    }
}
